esphome/components/sprinkler/sprinkler.cpp

```diff
@@ -954,10 +954,18 @@ void Sprinkler::pause() {
 }
 
 void Sprinkler::resume() {
-  if (this->paused_valve_.has_value() && (this->resume_duration_.has_value())) {
-    ESP_LOGD(TAG, "Resuming valve %u with %u seconds remaining", this->paused_valve_.value_or(0),
-             this->resume_duration_.value_or(0));
-    this->fsm_request_(this->paused_valve_.value(), this->resume_duration_.value());
+  if (this->standby()) {
+    ESP_LOGD(TAG, "resume called but standby is enabled; no action taken");
+    return;
+  }
+
+  if (this->paused_valve_.has_value()) {
```

Comment:
As a matter of form, I would probably leave this line intact:
```suggestion
  if (this->paused_valve_.has_value() && (this->resume_duration_.has_value())) {
```
...as you are otherwise assuming (below) that `resume_duration_.has_value()` is always true...which, it should be...but what if it isn't for some reason? 😇

Reply:
That said, I think there are some extra parentheses there.. 😅

Reply:
Totally makes sense, let's be explicit on conditions :)